collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -471,6 +467,14 @@ private void setupViewModels() {
         formEntryViewModel = new ViewModelProvider(this, formEntryViewModelFactory)
                 .get(FormEntryViewModel.class);
 
+        formEntryViewModel.getCurrentIndex().observe(this, index -> {
+            formIndexAnimationHandler.handle(index);
```

Comment:
I think you can convert `FormIndexAnimationHandler` to kotlin it's a small class and will be tested here so it should be safe.